cloud/scope/common.go

```diff
@@ -9,11 +9,14 @@ import (
 	"github.com/linode/linodego"
 	"golang.org/x/oauth2"
 	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api/util/patch"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/linode/cluster-api-provider-linode/version"
 )
 
+type patchNewHelper func(obj client.Object, crClient client.Client) (*patch.Helper, error)
```

Comment:
I missed this in an earlier PR. Is it necessary to mock `patch.Helper`? It already takes a `client.Client` interface (i.e. `k8sClient` interface), so it should be sufficient to just pass our generated mock to it and mock out the behavior we want.

Reply:
I tried that but it was getting much more complicated than it needed to be. `patch.NewHelper()` func's logic was asserting some checks which was causing it to be difficult to use `k8sClient` interface mock. Also the func calls `apiutil.GVKForObject(object interface{}, client.scheme())` which was also causing problems.

I'm sure we can figure it out with enough time but it didn't seem worth troubleshooting :(

Reply:
Updated!